db/install.xml

```diff
@@ -27,5 +27,25 @@
         <INDEX NAME="lockcounter" UNIQUE="false" FIELDS="userid, factor, lockcounter"/>
       </INDEXES>
     </TABLE>
+    <TABLE NAME="tool_mfa_secrets" COMMENT="Table to store factor secrets">
+      <FIELDS>
+        <FIELD NAME="id" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="true"/>
+        <FIELD NAME="userid" TYPE="int" LENGTH="10" NOTNULL="true" SEQUENCE="false"/>
+        <FIELD NAME="factor" TYPE="char" LENGTH="100" NOTNULL="true" SEQUENCE="false"/>
```

Comment:
should a secret be linked to a factor type or a factor instance id? id seems more correct to me. It might not make a different either way

Reply:
I cant think of a use case where the secret would need to be hardlinked to one specific factor instance. The factor type just confirms that this code is indeed for this factor, but it does not necessarily need to be hardlinked to a device for example. If there were 5 email addresses setup for example, all are a valid email code, regardless of the origin address that the code was received at. Happy to change it, but I cant see a use case for it. It is very convenient to Select * where type = email